src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -0,0 +1,681 @@
+    let result = null;
+    try {
+      result = await housing.saveApartmentApplication(applicationID, primaryUsername, applicants);
+    } catch {
+      result = false;
+    }
+    if (result !== null) {
+      console.log(result); //! DEBUG
+      this.setState({ applicationID: result, saving: 'success' });
+    } else {
+      this.snackbarText = 'Something went wrong while trying to save the application.';
+      this.snackbarSeverity = 'error';
+      this.setState({ snackbarOpen: true, saving: 'failed' });
+    }
+    if (this.saveButtonAlertTimeout === null) {
+      // Shows the success icon for 6 seconds and then returns back to normal button
+      this.saveButtonAlertTimeout = setTimeout(() => {
+        this.saveButtonAlertTimeout = null;
+        this.setState({ saving: false });
+      }, 6000);
+    }
+  }
+
+  handleCloseSnackbar = (event, reason) => {
+    // Prevent the snackbar from closing if the user clicks outside the snackbar
+    if (reason === 'clickaway') {
+      return;
+    }
+    this.setState({ snackbarOpen: false });
+  };
+
+  handleCloseDialog = (event, reason) => {
+    // Prevent the dialog box from closing if the user clicks outside the dialog box
+    if (reason === 'clickaway') {
+      return;
+    }
+    this.handleCloseOkay();
+  };
+
+  handleCloseOkay = () => {
+    this.setState({ submitDialogOpen: false, editDialogOpen: false });
+  };
+
+  render() {
+    if (this.props.authentication) {
+      /**
+       * Used to re-render the page when the network connection changes.
+       * this.state.network is compared to the message received to prevent
+       * multiple re-renders that creates extreme performance lost.
+       * The origin of the message is checked to prevent cross-site scripting attacks
+       */
+      window.addEventListener('message', (event) => {
+        if (
+          event.data === 'online' &&
+          this.state.network === 'offline' &&
+          event.origin === window.location.origin
+        ) {
+          this.setState({ network: 'online' });
+        } else if (
+          event.data === 'offline' &&
+          this.state.network === 'online' &&
+          event.origin === window.location.origin
+        ) {
+          this.setState({ network: 'offline' });
+        }
+      });
+
+      const primaryApplicantAlertText = (
+        <span>
+          If you change the primary applicant, you will no longer be able to edit this application
+          yourself.
+          <br />
+          Are you sure you want to change the primary applicant?
+        </span>
+      );
+
+      /**
+       * Gets status of current network connection for online/offline rendering
+       * Defaults to online in case of PWA not being possible
+       */
+      const networkStatus = JSON.parse(localStorage.getItem('network-status')) || 'online';
+
+      if (networkStatus === 'online' && this.props.authentication) {
+        return (
+          <div>
+            {this.state.loading ? (
```

Comment:
Should probably make this a separate if else, rather than burying it in the return of this if block. I think the flow would be more readable that way.